lib/vector-core/src/event/log_event.rs

```diff
@@ -150,7 +150,10 @@ impl LogEvent {
     /// valid for `LogNamespace::Legacy`
     pub fn from_str_legacy(msg: impl Into<String>) -> Self {
         let mut log = LogEvent::default();
-        log.insert(log_schema().message_key(), msg.into());
+        if let Some(message_key) = log_schema().message_key() {
+            log.insert((PathPrefix::Event, message_key), msg.into());
+        }
```

Comment:
This pattern to looks to be repeated all over. Would an insert helper be useful, maybe something like this?
```rust
    fn maybe_insert(&mut self, prefix: PathPrefix, path: OptionalValuePath, value: impl Into<Value>)
```
(the `impl Into` is to avoid calling `.into` unless the path is `Some`)

Reply:
Added `maybe_insert` function. Saved a few lines.